tests/python/relay/test_ad.py

```diff
@@ -0,0 +1,27 @@
+import tvm
+from tvm import relay
+from tvm.relay.ir_pass import fo_with_gradient
+from tvm.relay.op import log, add, equal, subtract, concat, multiply
+from tvm.relay.ir_builder import IRBuilder, convert
+
+def test_simple_gradient():
+    b = IRBuilder()
+    # f x = x * x + x
+    ty = relay.TensorType(tvm.convert([]), "float32")
+    x = relay.Var("x")
+    f = relay.Function([relay.Param(x, ty)], ty, multiply(x, x), [])
+    ret = fo_with_gradient(b.env, f)
```

Comment:
Do we need to emphasize "first-order" in the front API? Can we take `gradient` directly?

Reply:
we can, but after discussion with tianqi and jared, we think it is best to supply a family of ad algorithm with different characteristic